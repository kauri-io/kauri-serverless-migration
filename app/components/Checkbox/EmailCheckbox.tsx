import { Tooltip } from 'react-tippy'
import styled from 'styled-components'
import theme from '../../lib/theme-config'
import Checkbox, { IProps } from './index'

const containerWidth = 190

const TooltipContainer = styled.section`
    display: flex;
    flex-direction: column;
    align-items: center;
    background: white;
    position: relative;
    padding: ${theme.space[2]}px;
    width: ${containerWidth}px;
    text-align: center;
    > * {
        cursor: pointer;
    }
    > span:last-child {
        text-transform: uppercase;
        font-size: ${theme.fontSizes[0]}px;
        font-weight: ${theme.fontWeight[3]};
    }
    box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 4px;
    border-radius: 4px;
`

const TooltipArrow = styled.div`
    box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 4px;
    position: absolute;
    z-index: -1;
    bottom: -3%;
    width: 14px;
    height: 14px;
    background: white;
    transform: rotate(45deg);
    border-radius: 2px;
`

export const Content: React.SFC<{}> = () => (
    <TooltipContainer>
        <TooltipArrow />
        <span>
            Keep this checked to receive our newsletter with the latest
            tutorials and content series
        </span>
    </TooltipContainer>
)

const ReferenceContainer = styled.div`
    display: flex;
    width: 70px;
`

const Component: React.FunctionComponent<IProps> = props => (
    <ReferenceContainer>
        <Tooltip
            html={<Content />}
            position="top"
            trigger="mouseenter"
            unmountHTMLWhenHide={true}
        >
            <Checkbox
                label="newsletter"
                onChange={props.onChange}
                disabled={props.disabled}
                checked={props.checked}
            />
        </Tooltip>
    </ReferenceContainer>
)

export default Component
