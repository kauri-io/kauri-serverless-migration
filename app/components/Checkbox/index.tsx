import styled from 'styled-components'

const Styled = styled.div<{ disabled: boolean }>`
    display: flex;
    cursor: ${props => props.disabled && 'not-allowed'};
    > input {
        opacity: 0;
    }
    > input + label {
        position: relative;
        padding-left: 25px;
        cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
        color: white;
        text-transform: uppercase;
        font-size: 11px;
        font-weight: 700;
        &:before {
            content: '';
            position: absolute;
            left: 0;
            top: 0;
            width: 17px;
            height: 17px;
            border: 1px solid ${props => props.theme.primaryColor};
            background: transparent;
            border-radius: 4px;
        }
        &:after {
            content: url('/static/images/checked.svg');
            position: absolute;
            top: -4px;
            left: 4px;
            font-size: 16px;
            color: white;
            transition: all 0.2s;
        }
    }
    > input:not(:checked) + label {
        &:after {
            opacity: 0;
            transform: scale(0);
        }
    }
    > input:disabled:not(:checked) + label {
        &:before {
            background-color: transparent;
        }
    }
    > input:disabled:checked + label {
        &:after {
            opacity: 0.3;
        }
    }
    > input:disabled + label {
        opacity: 0.3;
    }
    > input:checked + label {
        &:after {
            opacity: 1;
            transform: scale(1);
        }
    }
    > input:checked:focus + label,
    input:not(:checked):focus + label {
        &:before {
            border: 1px dotted blue;
            border-radius: 4px;
        }
    }
`

export interface IProps {
    checked: boolean
    disabled: boolean
    onChange: (payload: any) => void
    label?: string
}

const Component: React.SFC<IProps> = props => (
    <Styled
        disabled={props.disabled}
        onClick={() => !props.disabled && props.onChange(!props.checked)}
    >
        <input
            disabled={props.disabled}
            type="checkbox"
            checked={props.checked}
        />
        <label>{props.label}</label>
    </Styled>
)

export default Component
