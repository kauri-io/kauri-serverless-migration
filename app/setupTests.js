import Enzyme, { shallow, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import theme from './lib/theme-config'
import { ThemeProvider } from 'styled-components'

Enzyme.configure({ adapter: new Adapter() })

const observeMock = {
    observe: () => null,
    disconnect: () => null, // maybe not needed
}

window.IntersectionObserver = () => observeMock

export const mountWithTheme = children =>
    mount(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const renderWithTheme = children =>
    render(<ThemeProvider theme={theme}>{children}</ThemeProvider>)

export const shallowWithTheme = children =>
    shallow(<ThemeProvider theme={theme}>{children}</ThemeProvider>)
