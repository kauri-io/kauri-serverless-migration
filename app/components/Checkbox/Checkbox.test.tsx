import Checkbox from './index'
import { mountWithTheme } from '../../setupTests'

describe('components/Checkbox', () => {
    it('snapshot should match with checked true and disabled false', () => {
        const props = {
            checked: true,
            disabled: false,
            onChange: (e: boolean) => e,
        }
        const wrapper = mountWithTheme(<Checkbox {...props} />)
        expect(wrapper).toMatchSnapshot()
    })

    it('snapshot should match with checked false and disabled true', () => {
        const props = {
            checked: false,
            disabled: true,
            onChange: (e: boolean) => e,
        }
        const wrapper = mountWithTheme(<Checkbox {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})
