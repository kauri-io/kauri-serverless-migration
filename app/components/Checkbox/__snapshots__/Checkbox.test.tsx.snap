// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Checkbox should match snapshot with checked false and disabled true 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`components/Checkbox should match snapshot with checked true and disabled false 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
